### PR TITLE
BUG: GH38672 SeriesGroupBy.value_counts for categorical

### LINKED ISSUE #38672: BUG: SeriesGroupBy.value_counts is inconsistent with Series.value_counts when the Series is categorical
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import pandas as pd

df = pd.DataFrame()
df["col1"] = ["A", "A", "B", "B"]
df["col2"] = pd.Categorical(["1", "2", "1", "1"], categories=["1", "2", "3"])

print(df.groupby("col1").col2.value_counts())

```

#### Problem description

A *value_counts* applied to the col2 Series:
```python
df.col2.value_counts()
```

 yields:
```
1      3
2      1
3      0
Name: col2, dtype: int64
```
Because col2 is categorical we get the 0 count for value 3, as desired.

If we groupby col1 first 
```python
df.groupby("col1").col2.value_counts()
```
we get the following output:
```
col1  col2
A     1       1
      2       1
B     1       2
Name: col2, dtype: int64
```

#### Expected Output

The expected output can be achieved with a workaround:
```python
df.groupby("col1").col2.apply(pd.Series.value_counts)
```

which yields:
```
col1   
A     2    1
      1    1
      3    0
B     1    2
      3    0
      2    0
Name: col2, dtype: int64
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 67a3d4241ab84419856b84fc3ebc9abcbe66c6b3
python           : 3.7.5.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.18362
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : None.None

pandas           : 1.1.4
numpy            : 1.19.1
pytz             : 2019.3
dateutil         : 2.8.0
pip              : 20.2.3
setuptools       : 41.2.0
Cython           : 0.29.21
pytest           : 5.3.5
hypothesis       : None
sphinx           : 2.4.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : 4.4.1
html5lib         : 1.0.1
pymysql          : 0.9.3
psycopg2         : None
jinja2           : 2.10.3
IPython          : 7.8.0
pandas_datareader: None
bs4              : 4.8.2
bottleneck       : None
fsspec           : 0.6.0
fastparquet      : None
gcsfs            : None
matplotlib       : 3.1.3
numexpr          : None
odfpy            : None
openpyxl         : 3.0.0
pandas_gbq       : None
pyarrow          : 1.0.1
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.4.1
sqlalchemy       : 1.3.10
tables           : None
tabulate         : 0.8.6
xarray           : None
xlrd             : 1.2.0
xlwt             : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by venaturum**
Unobserved categories in Series were being dropped in value_counts, which was inconsistent with Series.value_counts

- [x] closes #38672
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/groupby/test_value_counts.py [hunk 1]
```diff
@@ -85,8 +85,10 @@ def rebuild_index(df):
     tm.assert_series_equal(left.sort_index(), right.sort_index())
 
 
-def test_series_groupby_value_counts_with_grouper():
+@pytest.mark.parametrize("categorical", [True, False])
```
Comment (jreback) [inline pandas/tests/groupby/test_value_counts.py:88]: make another test as we want to use the observed fixture.

Comment (jreback) [reply pandas/tests/groupby/test_value_counts.py:88]: pls also hard code the expected results for both cases (much easier to see / understand).

I would also expected this change to break something, aren't the current expected results wrong? or are we just not testing them.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/groupby/test_value_counts.py (hunk 1 above)
Comment (venaturum) [reply pandas/tests/groupby/test_value_counts.py:88]: Thanks @jreback 

Current results aren't tested at the moment. The proposed changes to the test file without the corresponding changes to pandas/core/groupby/generic.py will result in a failed assertion. 

I am unsure about what your comment re observed fixture means, so just to clarify, I should :

1) roll back my changes to test_value_counts.py

2) leave existing function test_series_groupby_value_counts_with_grouper as it was

3) create new test for when the Series giving rise to the SeriesGroupBy is categorical and hardcode expected results

I'm not sure what "both cases" is referring to. Unless you're intending me to rewrite the existing function test_series_groupby_value_counts_with_grouper to use hardcoded expected results also?

Comment (jreback) [reply pandas/tests/groupby/test_value_counts.py:88]: yes, simply use the observed fixed as an argument to groupby that's 2 cases to test. i wonder if these give the same result.

Comment (venaturum) [reply pandas/tests/groupby/test_value_counts.py:88]: I'm happy to write a test where the groupby has observed=True, however isn't it only relevant if the groupers are categorical, which is not the case here?

Comment (jreback) [reply pandas/tests/groupby/test_value_counts.py:88]: hmm actually yeah observed is not relevant. ok pls make a new test in any event.

**Comment by jreback** [comment]
also pls add a whatsnew note to 1.3, bug fixes under categorical is fine.

so i guess this was just not tested before.

**Comment by venaturum** [comment]
@jreback

> also pls add a whatsnew note to 1.3, bug fixes under categorical is fine.

No worries.  I think it might be more consistent with past whatsnew files to place it under Groupby/resample/rolling though.

Happy to do what I'm told though 😄

File: pandas/core/groupby/generic.py [hunk 2]
```diff
@@ -681,9 +682,7 @@ def value_counts(
         from pandas.core.reshape.merge import get_join_indexers
         from pandas.core.reshape.tile import cut
 
-        if bins is not None and not np.iterable(bins):
-            # scalar bins cannot be done at top level
-            # in a backward compatible way
+        def apply_series_value_counts():
```
Comment (jreback) [inline pandas/core/groupby/generic.py:685]: I would move this downto right before L 697 as well

**Review by jreback — CHANGES_REQUESTED** [review]

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @venaturum

**Comment by jreback** [comment]
might resolve https://github.com/pandas-dev/pandas/issues/32471 or https://github.com/pandas-dev/pandas/issues/25970

**Pull request merged**